synapse/handlers/sync.py

```diff
@@ -277,8 +277,9 @@ async def wait_for_sync_for_user(
         user_id = sync_config.user.to_string()
         await self.auth.check_auth_blocking(requester=requester)
 
-        res = await self.response_cache.wrap(
+        res = await self.response_cache.wrap_conditional(
             sync_config.request_key,
+            lambda result: since_token != result.next_batch,
```

Comment:
this should have had a comment. Why does doing this give the right behaviour? (I shouldn't have to go and find the PR that added it, and even having done so it's hard to correlate the description to the code.)

Reply:
It was described in the PR, but i'll take it in another PR to add a comment here, i.e. "small fixes for `ResponseCache`"

Reply:
> It was described in the PR description,

yeah as I said: having to go and find the PR and grok it is a pain.

Reply:
Alright, i'll add a comment describing what that conditional does :+1: